safe_transaction_service/history/services/transaction_service.py

```diff
@@ -371,3 +373,546 @@ def serialize_all_txs_v2(
 
         logger.debug("Serialized all transactions")
         return results
+
+    def get_export_transactions(
+        self,
+        safe_address: str,
+        execution_date_gte: Optional[datetime] = None,
+        execution_date_lte: Optional[datetime] = None,
+        limit: int = 1000,
+        offset: int = 0,
+    ) -> Tuple[List[Dict[str, Any]], int]:
+        """
+        Get transactions optimized for CSV export using raw SQL queries
+
+        :param safe_address: Safe address to get transactions for
+        :param execution_date_gte: Filter transactions executed after this date
+        :param execution_date_lte: Filter transactions executed before this date
+        :param limit: Maximum number of transactions to return
+        :param offset: Number of transactions to skip
+        :return: Tuple of (transactions, total_count)
+        """
+        logger.debug(
+            "[%s] Getting export transactions with raw SQL: gte=%s, lte=%s, limit=%d, offset=%d",
+            safe_address,
+            execution_date_gte,
+            execution_date_lte,
+            limit,
+            offset,
+        )
+
+        # Base WHERE conditions for the final SELECT
+        where_conditions = []
+        params = []
+
+        if execution_date_gte:
+            where_conditions.append("execution_date >= %s")
```

Comment:
Maybe you should assert that `execution_date_gte` is really a datetime, as it can lead to sql injection if called with a string from other part of the code

Reply:
Done :)